Core.ApplicationServices/References/ReferenceService.cs

```diff
@@ -100,6 +102,45 @@ public class ReferenceService : IReferenceService
                     onNone: () => new OperationError("Root entity could not be found", OperationFailure.NotFound)
                 );
         }
+        public Result<ExternalReference, OperationError> UpdateReference(
+            int rootId,
+            ReferenceRootType rootType,
+            Guid referenceUuid,
+            ExternalReferenceProperties externalReferenceProperties)
+        {
+            return _referenceRepository
+                .GetRootEntity(rootId, rootType)
+                .Match
+                (
+                    onValue: root =>
+                    {
+                        if (!_authorizationContext.AllowModify(root))
+                        {
+                            return new OperationError("Not allowed to modify root entity", OperationFailure.Forbidden);
+                        }
+
+                        return _referenceRepository.GetByUuid(referenceUuid)
+                            .Match(reference =>
+                            {
+                                reference.Title = externalReferenceProperties.Title;
+                                reference.ExternalReferenceId = externalReferenceProperties.DocumentId;
+                                reference.URL = externalReferenceProperties.Url;
+                                reference.Created = _operationClock.Now;
+
+                                if (externalReferenceProperties.MasterReference)
+                                {
+                                    root.SetMasterReference(reference);
+                                }
+                                RaiseRootUpdated(root);
```

Comment:
looks like the other methods raise domain events on created/removed.. you can raise one for updated on the reference before raising on the root